app/services/google/parser_service.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+module Google
+  class ParserService
+    NON_ADS_RESULT_SELECTOR = 'a[data-ved]:not([role]):not([jsaction]):not(.adwords):not(.footer-links)'
+    AD_CONTAINER_ID = 'tads'
+    ADWORDS_CLASS = 'adwords'
+
+    def initialize(html_response:)
+      raise ArgumentError, 'response.body cannot be blank' if html_response.body.blank?
+
+      @html = html_response
+
+      @document = Nokogiri::HTML.parse(html_response)
+
+      # Add a class to all AdWords link for easier manipulation
+      document.css('div[data-text-ad] a[data-ved]').add_class(ADWORDS_CLASS)
+
+      # Mark footer links to identify them
+      document.css('#footcnt a').add_class('footer-links')
```

Comment:
In this scope, we should better not raise any errors or customize the `document`, just initialize some values that we need before moving on.

```suggestion
      @html = html_response

      @document = Nokogiri::HTML.parse(html_response) if html_response.body
```